src/addon-manager.js

```diff
@@ -369,6 +368,58 @@ class AddonManager extends EventEmitter {
     return hash.digest('hex');
   }
 
+  /**
+   * @method validateManifestObject
+   *
+   * Verifies one level of an object, and recurses as required.
+   */
+  validateManifestObject(prefix, object, template) {
+    for (let key in template) {
+      if (key in object) {
+        let objectVal = object[key];
+        let templateVal = template[key];
+        if (typeof(objectVal) != typeof(templateVal)) {
+          return `Expecting ${prefix}${key} to have type: ` +
+                 typeof(templateVal) + ', found: ' + typeof(objectVal);
+        }
+        if (typeof(objectVal) === 'object') {
+          let err = this.validateManifestObject(prefix + key + '.',
+                                                objectVal, templateVal);
+          if (err) {
+            return err;
+          }
+        }
+      } else {
+        return `Manifest is missing: ${prefix}${key}`;
+      }
+    }
+  }
+
+  /**
+   * @method validateManifest
+   *
+   * Verifies that the manifest looks valid. We only need to validate
+   * fields that we actually use.
+   */
+
+  validateManifest(manifest) {
+    let manifestTemplate = {
+      name: '',
```

Comment:
We should probably require `version` as well, in preparation for handling add-on updates. I will take care of adding support for the `files` array.